plugin/db/mysql/restore.go

```diff
@@ -526,22 +526,13 @@ func (driver *Driver) downloadBinlogFilesOnServer(ctx context.Context, binlogFil
 		if fileOnServer.Name == latestBinlogFileOnServer.Name && !downloadLatestBinlogFile {
 			continue
 		}
-		fileLocal, existLocal := binlogFilesLocalMap[fileOnServer.Name]
+		_, existLocal := binlogFilesLocalMap[fileOnServer.Name]
 		path := filepath.Join(driver.binlogDir, fileOnServer.Name)
 		if !existLocal {
 			if err := driver.downloadBinlogFile(ctx, fileOnServer, fileOnServer.Name == latestBinlogFileOnServer.Name); err != nil {
 				log.Error("Failed to download binlog file", zap.String("path", path), zap.Error(err))
 				return errors.Wrapf(err, "failed to download binlog file %q", path)
 			}
```

Comment:
As we don't use the tmp swap mechanism to download the file, we may get an incomplete file.

Reply:
True. I'll do the tmp swap in this PR.